cmd/gen_markdown.go

```diff
@@ -30,7 +40,56 @@ func newGenMarkdownCmd(root *cobra.Command) *cobra.Command {
 		Short:  "Generate Pulumi CLI documentation as Markdown (one file per command)",
 		Hidden: true,
 		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
-			return doc.GenMarkdownTree(root, args[0])
+			var files []string
+
+			// filePrepender is used to add front matter to each file, and to keep track of all
+			// generated files.
+			filePrepender := func(s string) string {
+				// Keep track of the generated file.
+				files = append(files, s)
+
+				// Add some front matter to each file.
+				fileNameWithoutExtension := strings.TrimSuffix(filepath.Base(s), ".md")
+				title := strings.Replace(fileNameWithoutExtension, "_", " ", -1)
+				alias := fileNameWithoutExtension + ".html"
+				buf := new(bytes.Buffer)
+				buf.WriteString("---\n")
+				buf.WriteString(fmt.Sprintf("title: \"%s\"\n", title))
```

Comment:
You can replace `\"%s\"` with just `%q`, and go will print the string with quotations for you.

Reply:
Nice! Done.